swayfmt/src/utils/language/expr/asm_block.rs

```diff
@@ -140,20 +140,37 @@ impl Format for AsmRegisterDeclaration {
     }
 }
 
+impl Format for Instruction {
+    fn format(
+        &self,
+        formatted_code: &mut FormattedCode,
+        _formatter: &mut Formatter,
+    ) -> Result<(), FormatterError> {
+        write!(formatted_code, "{:<4}", &self.op_code_ident().as_str())?;
+        for arg in self.register_arg_idents() {
+            write!(formatted_code, " {}", arg.as_str())?
+        }
+        if let Some(ident) = self.immediate_ident_opt() {
+            write!(formatted_code, " {}", ident.as_str())?
+        }
```

Comment:
Is this assuming that `asm` instructions always look like `<opcode> <r> <r> .. <i>`, so a list of registers followed by at most 1 immediate? While that may be true today, I find that a bit limiting? Can we not simply loop over all the args and write them regardless of their type?

Reply:
Good point! I took a stab at it